laser_encoders/language_list.py

```diff
@@ -13,6 +13,34 @@
 # -------------------------------------------------------
 # Language mapping to handle different language codes and names
 
+
+def build_language_names_dict(language_list, language_names):
+    result_dict = {}
+
+    for lang_code in language_list:
+        if lang_code in language_names:
+            names_list = language_names[lang_code]
+
+            # Check if the names_list is a list or a string
+            if isinstance(names_list, list):
+                for name in names_list:
+                    if name not in result_dict:
+                        result_dict[name] = []
+                    result_dict[name].append(lang_code)
+            else:
```

Comment:
Maybe, you could simplify the code by dropping this `else` branch and adding the following code above?
```
if not isinstance(names_list, list):
    names_list = [names_list]
```